src/MyInfoConnector/MyInfoConnectorConfig.cs

```diff
@@ -0,0 +1,233 @@
+        {
+            if (_x509private == null)
+            {
+                var filePath = ResolvePath(PrivateCertificateFilename);
+                _x509private = new X509Certificate2(filePath, PrivateCertificatePassword, X509KeyStorageFlags.MachineKeySet | X509KeyStorageFlags.Exportable);
+            }
+            var privateKey = _x509private.PrivateKey as RSACryptoServiceProvider;
+            return privateKey;
+        }
+
+
+        internal RSA GetPublicKey()
+        {
+            if (_x509public == null)
+            {
+                var filePath = ResolvePath(PublicCertificateFilename);
+                _x509public = new X509Certificate2(filePath);
+            }
+            var publicKey = _x509public.GetRSAPublicKey();
+            return publicKey;
+        }
+
+
+        private static string ResolvePath(string filename)
+        {
+            var filePath = filename;
+            if (!File.Exists(filename))
+            {
+                // It must be a relative path from bin folder
+                var appFolder = GetApplicationFolder();
+                filePath = Path.Combine(appFolder, filename);
+            }
+            return filePath;
+        }
+
+        private static string GetApplicationFolder()
+        {
+            var thisAssembly = Assembly.GetExecutingAssembly();
+
+            var assemblyPath = thisAssembly.Location;
+
+            var folder = Path.GetDirectoryName(assemblyPath);
+
+            if (folder.ToLower().Contains("windows"))
+            {
+                // Handle asp.net running from a temp folder in C:\Windows
+                return AppDomain.CurrentDomain.BaseDirectory + "bin";
+            }
+
+            return folder;
+        }
+
+        /// <summary>
+        /// Sanity check the settings but don't throw
+        /// Allow the caller to react appropriately
+        /// </summary>
+        public (bool isValid, string[] messages) IsValid()
+        {
+            var messages = new List<string>();
+
+            if (string.IsNullOrEmpty(PrivateCertificateFilename)) messages.Add($"PrivateCertificateFilename missing or empty");
+            if (string.IsNullOrEmpty(PrivateCertificatePassword)) messages.Add($"PrivateCertificatePassword missing or empty");
+            if (string.IsNullOrEmpty(PublicCertificateFilename)) messages.Add($"PublicCertificateFilename missing or empty");
+            if (string.IsNullOrEmpty(ClientAppId)) messages.Add($"ClientAppId missing or empty");
+            if (string.IsNullOrEmpty(ClientAppPassword)) messages.Add($"ClientAppPassword missing or empty");
+            if (string.IsNullOrEmpty(AttributeCsv)) messages.Add($"AttributeCsv missing or empty");
+            if (string.IsNullOrEmpty(Environment)) messages.Add($"Environment missing or empty");
+            if (string.IsNullOrEmpty(TokenUrl)) messages.Add($"TokenUrl missing or empty");
+            if (string.IsNullOrEmpty(PersonUrl)) messages.Add($"PersonUrl missing or empty");
```

Comment:
Since these are static string without variables, `$` seems redundant.
```suggestion
            if (string.IsNullOrEmpty(PrivateCertificateFilename)) messages.Add("PrivateCertificateFilename missing or empty");
            if (string.IsNullOrEmpty(PrivateCertificatePassword)) messages.Add("PrivateCertificatePassword missing or empty");
            if (string.IsNullOrEmpty(PublicCertificateFilename)) messages.Add("PublicCertificateFilename missing or empty");
            if (string.IsNullOrEmpty(ClientAppId)) messages.Add("ClientAppId missing or empty");
            if (string.IsNullOrEmpty(ClientAppPassword)) messages.Add("ClientAppPassword missing or empty");
            if (string.IsNullOrEmpty(AttributeCsv)) messages.Add("AttributeCsv missing or empty");
            if (string.IsNullOrEmpty(Environment)) messages.Add("Environment missing or empty");
            if (string.IsNullOrEmpty(TokenUrl)) messages.Add("TokenUrl missing or empty");
            if (string.IsNullOrEmpty(PersonUrl)) messages.Add("PersonUrl missing or empty");
```